nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/NiFiCsrfTokenRepository.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.web;
+
+
+import org.springframework.security.web.csrf.CsrfToken;
+import org.springframework.security.web.csrf.CsrfTokenRepository;
+import org.springframework.security.web.csrf.DefaultCsrfToken;
+import org.springframework.util.Assert;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.web.util.WebUtils;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.reflect.Method;
+
+/**
+ * A {@link CsrfTokenRepository} implementation for NiFi that matches the NiFi Cookie JWT against the
+ * Authorization header JWT to protect against CSRF. If the request is an idempotent method type, then only the Cookie
+ * is required to be present - this allows authenticating access to static resources using a Cookie. If the request is a non-idempotent
+ * method, NiFi requires the Authorization header (eg. for POST requests).
+ *
+ * @author Rob Winch
+ * @since 4.1
+ */
+public final class NiFiCsrfTokenRepository implements CsrfTokenRepository {
```

Comment:
Though I understand you got this implementation from something like [CookieCsrfTokenRepository](https://github.com/spring-projects/spring-security/blob/master/web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java), I wonder if it would be preferable to use a [Delegation pattern ](https://refactoring.guru/replace-inheritance-with-delegation)here instead of reimplementing CookieCsrfTokenRepository.

Reply:
Let me know what you think of this now with my update.

Reply:
Perfect!